content/docs/guides/selfbuild/bring-your-own-llvm.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "Build with system-installed LLVM"
+weight: 3
+description: >
+  How to build TinyGo with a system-installed version of LLVM.
+---
+
+⚠️ Halt! This is the system installed LLVM guide! Please check the following table and make sure you don't need the [manual LLVM install guide](./manual-llvm) instead! 
+
+
+| You need to build LLVM manually in the following cases |
+|---|
+|  You would like to use it for the ESP8266 or ESP32 chips |
+| You are using Windows. |
+| Your Linux distribution (if you use Linux) does not ship the right LLVM version. |
+
+
+Using a system-installed version of LLVM depends on your system, of course.
+
+For **Debian** or **Ubuntu** you can install LLVM by adding a new apt repository. For more information about this method, see [apt.llvm.org](https://apt.llvm.org/). *Before copying the command below, please replace `xxxxx` with your distribution's codename*.
+
+| Distro | Version | Codename |
+|--------|------- |-----------|
+| Ubuntu | 18.04  | `bionic`  |
+| Ubuntu | 20.04  | `focal`   |
+| Ubuntu | 20.10  | `groovy`  |
+| Ubuntu | 21.04  | `hirsute` |
+| Ubuntu | 22.04  | `jammy`   |
+| Debian | 10     | `buster`  |
+| Debian | 11     | `bullseye`|
+| Debian | sid    | `unstable`|
+
+```shell
+echo 'deb http://apt.llvm.org/xxxxx/ llvm-toolchain-xxxxx-16 main' | sudo tee /etc/apt/sources.list.d/llvm.list
+```
+
+After adding the apt repository for your distribution you may install the LLVM toolchain packages:
+
+```shell
+wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+sudo apt-get update
+sudo apt-get install clang-16 llvm-16-dev lld-16 libclang-16-dev
+```
+
+For **MacOS**, you can install LLVM through [Homebrew](https://formulae.brew.sh/formula/llvm). The Clang/LLVM version from Apple is not supported by TinyGo.
+
+```shell
+brew install llvm
+```
+
+For **Fedora** users you can install LLVM from the repository. Note that the version of LLVM [varies by Fedora version](https://packages.fedoraproject.org/pkgs/llvm/llvm-libs/), for example Fedora 37 has LLVM 15.
+
+```shell
+sudo dnf install llvm-devel lld-libs lld
+```
+
+After LLVM has been installed, installing TinyGo should be as easy as running the following command:
+
+```shell
+go install
+```
+
+You should now have a working TinyGo installation!
```

Comment:
I wondered after "go install" was finished, that "tinygo version" is not working. So IMO the "Final notes" section should be referenced here or:
* move this concluding remarks to the end
* make the "Final notes" section an optional or checking step